src/webui/api/torrentscontroller.cpp

```diff
@@ -1475,6 +1475,26 @@ void TorrentsController::addTagsAction()
     }
 }
 
+void TorrentsController::setTagsAction()
+{
+    requireParams({u"hashes"_s, u"tags"_s});
+
+    const QStringList hashes {params()[u"hashes"_s].split(u'|', Qt::SkipEmptyParts)};
+    const QStringList tags {params()[u"tags"_s].split(u',', Qt::SkipEmptyParts)};
+
+    TagSet newTags {tags.begin(), tags.end()};
+    applyToTorrents(hashes, [newTags](BitTorrent::Torrent *const torrent)
+    {
+        for (const Tag &tag : asConst(torrent->tags()))
+        {
+            if (newTags.erase(tag) == 0)
+                torrent->removeTag(tag);
+        }
+        for (const Tag &tag : newTags)
+            torrent->addTag(tag);
+    });
```

Comment:
Suggestion to make it compile:
```suggestion
    const TagSet newTags {tags.begin(), tags.end()};
    applyToTorrents(hashes, [&newTags](BitTorrent::Torrent *const torrent)
    {
        TagSet tmpTags {newTags};
        for (const Tag &tag : asConst(torrent->tags()))
        {
            if (tmpTags.erase(tag) == 0)
                torrent->removeTag(tag);
        }
        for (const Tag &tag : tmpTags)
            torrent->addTag(tag);
    });
```

Reply:
Thanks!! This compiles and looks to be working as expected. The previous code before before the removal of just the newTags ref did also work in my testing locally, but this is probably better.

Reply:
> Suggestion to make it compile

👍 
I just discovered that it doesn't compile, and I realized my mistake, but I didn't have time to offer the correct code, you beat me to it.

Reply:
> The previous code before before the removal of just the newTags ref did also work in my testing locally, but this is probably better.

I believe you missed to test affected case.

Reply:
Most likely. My local test instance isn't that big, so probably didn't hit that case. Nice find :clap: